OrbitGl/TrackManager.cpp

```diff
@@ -0,0 +1,449 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "TrackManager.h"
+
+#include <GteVector.h>                     // for Vector
+#include <absl/container/flat_hash_map.h>  // for flat_hash_map
+#include <absl/strings/ascii.h>            // for AsciiStrToLower
+#include <absl/strings/match.h>            // for StrContains
+#include <absl/strings/str_format.h>       // for StrFormat
+#include <absl/strings/str_split.h>        // for SkipWhitespace, StrSplit, Splitter
+
+#include <algorithm>    // for find, min
+#include <cstdint>      // for int32_t, uint64_t, uint32_t
+#include <optional>     // for optional
+#include <string_view>  // for string_view
+#include <utility>      // for pair, move
+
+#include "App.h"
+#include "CoreMath.h"  // for Vec2
+#include "CoreUtils.h"
+#include "GlCanvas.h"  // for GlCanvas, GlCanvas::kZOffsetMovingTack, GlCanvas::kZOffsetPinnedT...
+#include "OrbitBase/Logging.h"  // for CHECK
+#include "OrbitBase/ThreadConstants.h"
+#include "OrbitClientData/CallstackData.h"  // for CallstackData
+#include "OrbitClientModel/CaptureData.h"   // for CaptureData
+#include "TimeGraph.h"                      // for TimeGraph
+#include "TimeGraphLayout.h"                // for TimeGraphLayout
+
+using orbit_client_protos::FunctionInfo;
+
+TrackManager::TrackManager(TimeGraph* time_graph) : time_graph_(time_graph) {
+  GetOrCreateSchedulerTrack();
+
+  tracepoints_system_wide_track_ = GetOrCreateThreadTrack(orbit_base::kAllThreadsOfAllProcessesTid);
+}
+
+void TrackManager::Clear() {
+  tracks_.clear();
+  scheduler_track_ = nullptr;
+  thread_tracks_.clear();
+  gpu_tracks_.clear();
+  graph_tracks_.clear();
+  async_tracks_.clear();
+  frame_tracks_.clear();
+
+  sorted_tracks_.clear();
+  sorted_filtered_tracks_.clear();
+
+  GetOrCreateSchedulerTrack();
+  tracepoints_system_wide_track_ = GetOrCreateThreadTrack(orbit_base::kAllThreadsOfAllProcessesTid);
+}
+
+void TrackManager::SetStringManager(std::shared_ptr<StringManager> str_manager) {
+  string_manager_ = std::move(str_manager);
+}
+
+std::vector<Track*> TrackManager::GetTracks() const {
+  std::vector<Track*> tracks;
+  for (auto track : tracks_) {
+    tracks.emplace_back(track.get());
```

Comment:
As far as I know, there is no need to use emplace here, as the data_type is a primitive one anyways. So consider just use `push_back` here.